src/types/File.ts

```diff
@@ -0,0 +1,40 @@
+import { ajv } from '../ajv';
+import type { Uuid } from './Uuid';
+import type { KeycloakId } from './KeycloakId';
+import type { JSONSchemaType } from 'ajv';
+import type { PresignedPost } from '@aws-sdk/s3-presigned-post';
+import type { Writable } from './Writable';
+
+interface File {
+	readonly id: number;
```

Comment:
Why not use `storageKey` here? I suppose the integer `id` provides insert order? But we have a `createdAt` that can be used.

Reply:
Suppose the (eventual) URL is the key. What bad thing happens then?

Reply:
That's poor design -- the key should not be based on our back end storage implementations.

Primary keys should be identifiers, not attributes.

URI is an attribute of the file, not an identifier of the file.